src/helpers/logging-interceptor.helper.ts

```diff
@@ -0,0 +1,19 @@
+import { CallHandler, ExecutionContext, Injectable, Logger, NestInterceptor } from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+@Injectable()
+export class LoggingInterceptor implements NestInterceptor {
+  private readonly logger = new Logger();
+
+  intercept(context: ExecutionContext, next: CallHandler): Observable<any> {
+    const requestBody = context.switchToHttp().getRequest().body;
+    this.logger.debug(`Request body: ${JSON.stringify(requestBody)}`);
+
+    return next.handle().pipe(
+      tap((responseBody) => {
+        this.logger.debug(`Response body: ${JSON.stringify(responseBody)}`);
+      }),
```

Comment:
The pino logger is usually used by logging an object alongside a message, rather than a message directly.
For example, our `AxiosRequestInterceptor`s log like so:
```
logger.debug({ outgoingRequest: { /* the request */ }, 'Sending the following HTTP request.');
```

This approach of stringifying the data and logging it as a message directly is not consistent with that - do you think it's better?

Reply:
I do think the messages need to be more precise if we do keep this format of logging, e.g. something like `Handling request with body: ` instead of `Request body: `, and `Returning response to client with body: ` instead of `Response body: `

Reply:
Having tried both ways I don't think my approach of logging is any better, I just wasn't aware of the existing pattern. I've changed my logs to

`this.logger.debug({ requestBody }, 'Handling the following request from the client.');`

and

`this.logger.debug({ responseBody }, 'Returning the following response to the client.');`

for consistency.